### PR TITLE
Add ability to parse gzipped requests and validate

### PR DESCRIPTION
**Pull request by cgearing**
This is an initial attempt at adding functionality for parsing gzipped bodies. 

Happy for comments.

### CODE REVIEW

File: flask_pydantic_spec/flask_backend.py [hunk 1]
```diff
@@ -140,7 +142,13 @@ def request_validation(
...
             req_query = {}
         if request.content_type and "application/json" in request.content_type:
-            parsed_body = request.get_json() or {}
+            if request.content_encoding and "gzip" in request.content_encoding:
+                raw_body = gzip.decompress(request.stream.read()).decode(
+                    encoding="utf-8"
+                )
+                parsed_body = json.loads(raw_body)
+            else:
+                parsed_body = request.get_json() or {}
```
Comment (SaurabhJha) [inline flask_pydantic_spec/flask_backend.py:151]: Where is `limit` checked?

**Review by SaurabhJha — APPROVED** [review]
looks good, one clarification.

File: flask_pydantic_spec/flask_backend.py (hunk 1 above)
Comment (cgearing) [reply flask_pydantic_spec/flask_backend.py:151]: `limit` is in the pydantic model

**Review by robyoung — APPROVED** [review]
Looks good. Is gzip what most clients will be sending?

**Comment by cgearing** [comment]
> Looks good. Is gzip what most clients will be sending?

I guess the most common alternative would be `brotli`, which could definitely be added in the future if required?
If we really wanted to future proof it I guess we could provide a way of providing an inflation function somehow to the validator class

**Pull request merged**